src/other/tables.jl

```diff
@@ -0,0 +1,39 @@
+using Tables, IteratorInterfaceExtensions
+
+Tables.istable(::Type{<:AbstractDataFrame}) = true
+Tables.columnaccess(::Type{<:AbstractDataFrame}) = true
+Tables.columns(df::AbstractDataFrame) = df
+Tables.rowaccess(::Type{DataFrame}) = true
+Tables.rows(df::DataFrame) = eachrow(df)
+
+Tables.schema(df::AbstractDataFrame) = Tables.Schema(names(df), eltypes(df))
+Tables.schema(df::DFRowIterator) = Tables.schema(df.df)
+
+getvector(x::AbstractVector) = x
+getvector(x) = collect(x)
+fromcolumns(x) = DataFrame(Any[getvector(c) for c in Tables.eachcolumn(x)], collect(propertynames(x)))
```

Comment:
Here and below in `DataFrame` method we do not provide `makeunique` argument. This means that if there are duplicate column names the constructor will fail in the future (and in general Tables.jl do not guarantee that column names are unique now if I understand the current design correctly).

Maybe this is what we want, but in other constructors we allow the user decide what to do with `makeunique` keyword argument.

Reply:
I'm open to suggestions here; how about we go w/ this for now and address feedback as it comes in?

Reply:
@nalimilan - any opinion? I can live with what we have now as adding `makeunique` keyword argument later is non breaking.

The only scenario when I find it is really relevant when you create a `DataFrame` from a file on a disk that has duplicate column names and want to pipe the data through some interface that supports Tables.jl interface and then you have no way to allow `DataFrame` creation with renamed columns using this API.

Reply:
It would indeed be nice to support `makeunique` as for other constructors. That should be really simple: `makeunique` just has to be passed to the lower-level `DataFrame` constructor.

Reply:
I've added support for `makeunique`.

Reply:
Let's discuss this at https://github.com/JuliaData/DataFrames.jl/pull/1308.